### PR TITLE
`Tutorial groups`: Fix an issue with the configuration

### PR DESCRIPTION
**Pull request by stefanwaldhauser**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

In the client code certain entities are sent to the server with an indirect connection to the tutorial group configuration. The problem was that the setters for the tutorial group configuration had a restriction that required dates to be in the 'uuuu-MM-dd' format. The Jackson library was used for deserialization of these entities, but it was failing because the indirectly connected configuration was sending dates in the full ISO string format, which is the default for the dayjs library. This mismatch between the expected date format and the actual date format was causing Jackson to fail the entire deserialization process. 

To fix this bug, the code was updated to move the date format checks out of the setters, allowing dates to be sent in the full ISO string format. This allowed Jackson to successfully deserialize the entities, and the problem was resolved.

### Steps for Testing
- One way to replicate the bug that was found by @ge65cer :
- Create a course and configure tutorial groups (create one or two) to make sure that the configuration exists in the database
- Reload the whole page to clear the client cache
- Now try to create a programming exercise. This should work without any problems now.

#### Exam Mode Testing
- Same as above but with an exam programming exercises

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2
#### Exam Mode Test
- [x] Test

### Test Coverage
Added tests that show the error is fixed now.

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/tutorialgroups/TutorialGroupScheduleIntegrationTest.java [hunk 1]
```diff
@@ -96,6 +96,35 @@ void createNewTutorialGroupWithSchedule_sessionFallsOnTutorialGroupFreeDay_shoul
         this.assertScheduledSessionIsCancelledOnDate(holidaySession, secondAugustMonday, persistedTutorialGroupId, persistedSchedule);
     }
 
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    void createNewTutorialGroupWithSchedule_wrongDateFormatInSchedule_shouldReturnBadRequest() throws Exception {
```
Comment (JohannesStoehr) [inline src/test/java/de/tum/in/www1/artemis/tutorialgroups/TutorialGroupScheduleIntegrationTest.java:101]: You could think about converting this to a @ParameterizedTest

**Review by JohannesStoehr — APPROVED** [review]
Code changes look good. Left one optional suggestion

File: src/test/java/de/tum/in/www1/artemis/tutorialgroups/TutorialGroupScheduleIntegrationTest.java (hunk 1 above)
Comment (stefanwaldhauser) [reply src/test/java/de/tum/in/www1/artemis/tutorialgroups/TutorialGroupScheduleIntegrationTest.java:101]: Not really needed imo as text exercise is just a standing for all other entities. The concrete entity that is indirectly connected to the configuration is not important here.

**Review by PhilipWerz — APPROVED** [review]
Everything worked as expected. Created tutorial groups and exercises. Tested on TS1

**Review by timo07k — APPROVED** [review]
Bug is fixed. Tested on TS1.

**Review by tobias-lippert — APPROVED** [review]
Tested on ts1.  Bug is fixed.

**Pull request merged**